samcli/lib/package/artifact_exporter.py

```diff
@@ -263,6 +263,17 @@ def export(self, resource_id, resource_dict, parent_dir):
             if temp_dir:
                 shutil.rmtree(temp_dir)
 
+    def get_property(self, resource_dict):
+        return jmespath.search(self.PROPERTY_NAME, resource_dict)
+
+    def is_pre_exported(self, template_dict):
+        property_value = self.get_property(template_dict if template_dict else {})
+
+        if not property_value and not self.PACKAGE_NULL_PROPERTY:
+            return False
+
+        return is_s3_url(property_value)
```

Comment:
what if self.PACKAGE_NULL_PROPERTY is set?

Reply:
we pass a None to is_s3_url which leads to a valuerror, rendering it false.